pkg/server/request_id_mw.go

```diff
@@ -0,0 +1,19 @@
+/*
+Copyright © 2023 The Spray Proxy Contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+package server
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+)
+
+// Middleware adding request ID to gin context
+func addRequestId() gin.HandlerFunc {
```

Comment:
I think we should add a note here if/when we want to adopt OpenTelemtry.

cc @amisstea.

```suggestion
// Middleware adding request ID to gin context
// Note that this is a simple unique ID that can be used for debugging purposes.
// In the future, this might be replaced with OpenTelemetry IDs/tooling.
func addRequestId() gin.HandlerFunc {
```

Reply:
Updated.